app/table-expression/src/autocomplete/index.ts

```diff
@@ -0,0 +1,52 @@
+import type { Completion, CompletionContext, CompletionResult } from '@codemirror/autocomplete'
+import { computed } from 'vue'
+import { completionTypeAt } from './completionType'
+
+/** Completion information for a method. */
+export interface MethodCompletionInfo {
+  name: string
+  description?: string | undefined
+}
+
+function getMethodOptions(infos: MethodCompletionInfo[]) {
+  const methods: Completion[] = []
+  const binaryOperators: Completion[] = []
+  const unaryOperators: Completion[] = []
+  for (const { name, description } of infos) {
+    if (/^[a-z]/.test(name)) {
+      methods.push({
+        label: name,
+        type: 'method',
+        ...(description ? { detail: description } : {}),
+        apply: name,
+      })
+    } else {
+      binaryOperators.push({ label: name, type: 'operator' })
+    }
+  }
+  return {
+    methods,
+    methodsWithParens: methods.map((method) => ({ ...method, apply: `${method.apply}(` })),
+    binaryOperators,
+    unaryOperators,
```

Comment:
unaryOperators is always empty, se perhaps it would be better to initialize it inline to keep it visible.

Reply:
It's kind of a stub, I'll delete it for now and deal with it when I implement builtins.